server/pulp/server/controllers/repository.py

```diff
@@ -30,6 +30,55 @@
 _logger = logging.getLogger(__name__)
 
 
+def get_associated_unit_ids(repo_id, unit_type, repo_content_unit_q=None):
+    """
+    Return a generator of unit IDs within the given repo that match the type and query
+
+    :param repo_id:     ID of the repo whose units should be queried
+    :type  repo_id:     str
+    :param unit_type:   ID of the unit type which should be retrieved
+    :type  unit_type:   str
+    :param repo_content_unit_q: any additional filters that should be applied to the
+                                RepositoryContentUnit search
+    :type  repo_content_unit_q: mongoengine.Q
+    :return:    generator of unit IDs
```

Comment:
Consider adding a space between arguments and return values. It makes a nice visual separation between inputs and output for readability.